src/main/kotlin/com/msd/robot/application/RobotApplicationService.kt

```diff
@@ -214,4 +220,213 @@ class RobotApplicationService(
+
+        val amountsByGroupedPlanet = resourceAmountRequestedPerPlanet(validMineCommands)
+
+        amountsByGroupedPlanet.forEach { (planet, amount) ->
+            mineResourcesOnPlanet(validMineCommands, planet, amount)
+        }
+    }
+
+    /**
+     * Create a map of planets and the resource each offers. Each planet either has a single resource or none,
+     * represented with a null value.
+     *
+     * @param mineCommands: A list of commands, for which the resources should be fetched.
+     * @return a map assigning each planet a resource or a null value
+     */
+    private fun getResourcesOnPlanets(mineCommands: List<MineCommand>) = mineCommands
+        .map {
+            try {
+                robotDomainService.getRobot(it.robotUUID).planet.planetId
+            } catch (rnfe: RobotNotFoundException) {
+                // we will handle this later, for we are just interested in the planets
+                null
+            } 
+        }
+        .filterNotNull()
+        .distinct()
+        .map {
+            it to try {
+                gameMapService.getResourceOnPlanet(it)
+            } catch (re: RuntimeException) {
+                // if there was any problem we just put null, this will cause an exception to be thrown later on
+                null
+            }
+        }
+        .filter { it.second != null }
+        .toList()
+        .toMap()
+
+    /**
+     * Creates a list of [ValidMineCommand]s, which represent valid [MineCommand]s but using the entity objects instead
+     * of their IDs. A MineCommand is valid, if
+     * 1. the specified robot UUID corresponds to an actual robot
+     * 2. the planet on which the robot is positioned exists and has a resource patch on it
+     * 3. the robot has the necessary mining level to mine the resource on the planet
+     *
+     * @param mineCommands: The list of [MineCommand]s which gets filtered for validity
+     * @param planetsToResources: A map of planets assigning each a resourceType or a null value, representing no
+     *                            resource present on the planet.
+     * @return a list of [ValidMineCommand]s, representing only the MineCommands which are valid and having replaced
+*              the IDs with the corresponding entities.
+     */
+    private fun replaceIdsByObjectsInValidMineCommands(
+        mineCommands: List<MineCommand>,
+        planetsToResources: Map<UUID, ResourceType?>
+    ): MutableList<ValidMineCommand> {
+        val validMineCommands = mutableListOf<ValidMineCommand>()
+
+        for (mineCommand in mineCommands) {
+            try {
+                val robot = robotDomainService.getRobot(mineCommand.robotUUID)
+                val resource = planetsToResources[robot.planet.planetId]
+                    ?: throw NoResourceOnPlanetException(robot.planet.planetId)
+                val extendedMineCommand = ValidMineCommand(
```

Comment:
Why is this variable named extendedMineCommand

Reply:
changed it